src/network/src/common/src/AbstractNetworkManager.cpp

```diff
@@ -20,4 +32,13 @@ bool AbstractNetworkManager::registerAgent(uint16_t agentID, uint32_t ip) {
         return true;
     }
     return false;
-}
\ No newline at end of file
+}
+
+uint16_t AbstractNetworkManager::getAgentList(uint16_t* agentList, uint8_t maxLength) const {
+    uint8_t index = 0;
+    std::tuple<uint16_t*, uint8_t, uint8_t> tuple = std::make_tuple(agentList, index, maxLength);
+    m_hashMap.forEach(
+        reinterpret_cast<bool (*)(const unsigned short&, const unsigned int&, void*)>(callback),
```

Comment:
Is it necessary to cast or it'sd possible to match/edit the signature to avoid casting?